### PR TITLE
ISPN-10601 fix continuous query in server-ng

### PR DESCRIPTION
**Pull request by karesti**
https://issues.jboss.org/browse/ISPN-10601

### CODE REVIEW

File: server/runtime/src/test/java/org/infinispan/server/functional/HotRodCacheContinuousQueries.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+   public static class Person {
+
+      @ProtoField(number = 1) public String name;
+
+      @ProtoField(number = 2) public Integer id;
+
+      public Person() {
+      }
+
+      public Person(String name, Integer id) {
```
Comment (ryanemerson) [inline server/runtime/src/test/java/org/infinispan/server/functional/HotRodCacheContinuousQueries.java]: FYI you can do now the following with the latest protostream release:

```java
      @ProtoField(number = 1)
      public String name;

      @ProtoField(number = 2)
      public Integer id;

      @ProtoFactory
      public Person(String name, Integer id) {
```

So there's no need for the no-arg constructor

Comment (karesti) [reply server/runtime/src/test/java/org/infinispan/server/functional/HotRodCacheContinuousQueries.java]: btw @ryanemerson these new marshalling works with quarkus ?

Comment (karesti) [reply server/runtime/src/test/java/org/infinispan/server/functional/HotRodCacheContinuousQueries.java]: wondering for here: https://github.com/infinispan-demos/harry-potter-quarkus/pull/8

Comment (wburns) [reply server/runtime/src/test/java/org/infinispan/server/functional/HotRodCacheContinuousQueries.java]: To answer simply: Yes

Comment (wburns) [reply server/runtime/src/test/java/org/infinispan/server/functional/HotRodCacheContinuousQueries.java]: https://github.com/quarkusio/quarkus/blob/master/integration-tests/infinispan-client/src/main/java/io/quarkus/it/infinispan/client/Author.java is an example

File: server/runtime/pom.xml [hunk 2]
```diff
@@ -117,6 +117,11 @@
          <artifactId>junit</artifactId>
          <scope>test</scope>
       </dependency>
+      <dependency>
```
Comment (anistor) [inline server/runtime/pom.xml]: So is this strictly for continuous query? Or query in general?

File: server/runtime/src/test/java/org/infinispan/server/functional/HotRodCacheContinuousQueries.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+/**
+ * @author Katia Aresti &lt;karesti@redhat.com&gt;
+ * @since 10.0
+ **/
+public class HotRodCacheContinuousQueries {
```
Comment (anistor) [inline server/runtime/src/test/java/org/infinispan/server/functional/HotRodCacheContinuousQueries.java]: We have some CQ tests in old server. I'd prefer to have those moved to the new server before adding other tests.

Comment (tristantarrant) [reply server/runtime/src/test/java/org/infinispan/server/functional/HotRodCacheContinuousQueries.java]: No, because they use entities, instead of proto.

**Comment by tristantarrant** [comment]
@karesti I have an actual fix coming up for registering extensions at boot time.

File: server/runtime/src/test/java/org/infinispan/server/functional/HotRodCacheContinuousQueries.java (hunk 3 above)
Comment (anistor) [reply server/runtime/src/test/java/org/infinispan/server/functional/HotRodCacheContinuousQueries.java]: RemoteContinuousQueryIT uses protobuf

Comment (anistor) [reply server/runtime/src/test/java/org/infinispan/server/functional/HotRodCacheContinuousQueries.java]: And besides CQ, we need to ensure filters+converters work in general. RemoteListenerWithDslFilterIT if testing that.

**Comment by tristantarrant** [comment]
Replaced by https://github.com/infinispan/infinispan/pull/7314

**Pull request closed without merging**